src/subsplit_dag.cpp

```diff
@@ -120,28 +120,43 @@ std::string SubsplitDAG::ToDot(bool show_index_labels) const {
   return string_stream.str();
 }
 
-const BitsetSizeMap &SubsplitDAG::GetGPCSPIndexer() const { return gpcsp_indexer_; }
-const BitsetSizePairMap &SubsplitDAG::GetParentToRange() const {
-  return parent_to_range_;
+BitsetSizeMap SubsplitDAG::GetGPCSPIndexer() const {
+  auto gpcsp_indexer = BitsetSizeMap();
+  ReversePostorderIndexTraversal([this, &gpcsp_indexer](size_t parent_id, bool rotated,
+                                                        size_t child_id,
+                                                        size_t gpcsp_idx) {
+    const auto parent_subsplit = GetDagNode(parent_id)->GetBitset(rotated);
+    const auto child_subsplit = GetDagNode(child_id)->GetBitset();
+    SafeInsert(gpcsp_indexer, Bitset::PCSPOfPair(parent_subsplit, child_subsplit),
+               gpcsp_idx);
+  });
+  for (const auto &rootsplit : rootsplits_) {
+    const auto subsplit = rootsplit + ~rootsplit;
+    SafeInsert(gpcsp_indexer, subsplit, GetRootsplitIndex(subsplit));
+  }
+  return gpcsp_indexer;
 }
 
 SubsplitDAGNode *SubsplitDAG::GetDagNode(const size_t node_id) const {
   return dag_nodes_.at(node_id).get();
 }
 
 size_t SubsplitDAG::GetRootsplitIndex(const Bitset &rootsplit) const {
-  return gpcsp_indexer_.at(rootsplit);
+  return RootsplitIndexOfIds(subsplit_to_id_.at(rootsplit));
+}
+
+size_t SubsplitDAG::RootsplitIndexOfIds(size_t root_id) const {
+  return dag_edges_.at({root_id, root_id});
```

Comment:
I think you plan on replacing this sort of thing later. In such a case I like to mark it with a comment with the associated issue number, but you don't have to if you don't want.